### PR TITLE
Issue #5811: Add CSpell action to core.

### LINKED ISSUE #5811: Add a GitHub Action to run cspell.
This is a sibling task to others, like #5245, #5296, and #5467.

Relevant information:
- CSpell: https://cspell.org | https://github.com/streetsidesoftware/cspell
- GitHub Action: https://github.com/marketplace/actions/cspell-action (there may be others to choose from)
- Issue where CSpell was added to D9 core: [Add .cspell.json to automate spellchecking in Drupal core](https://www.drupal.org/project/drupal/issues/2972224)
- Issue to add cspell to the DrupalCI: [Support code spell checking](https://www.drupal.org/project/drupalci/issues/3122096)
- The dictionary currently in use at the time of writing: https://git.drupalcode.org/project/drupal/-/blob/9.5.x/core/misc/cspell/dictionary.txt
- [Rename dictionary.txt file to drupal.dic to make it compatible with PhpStorm on Linux](https://www.drupal.org/project/drupal/issues/3153919)

### PR DESCRIPTION
**Pull request by ghost**
Fixes https://github.com/backdrop/backdrop-issues/issues/5811

### CODE REVIEW

File: .cspell/cspell.json [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  "$schema": "https://raw.githubusercontent.com/streetsidesoftware/cspell/main/cspell.schema.json",
+  "version": "0.2",
+  "dictionaryDefinitions": [
+    {
+      "name": "backdrop",
+      "path": "backdrop.dic"
+    },
+    {
+      "name": "drupal",
+      "path": "https://git.drupalcode.org/project/drupal/-/raw/10.1.x/core/misc/cspell/dictionary.txt"
```
Comment (klonos) [inline .cspell/cspell.json]: Can we please change this to be using the `HEAD` keyword, which gets the default branch every time?:
```suggestion
      "path": "https://git.drupalcode.org/project/drupal/-/raw/HEAD/core/misc/cspell/dictionary.txt"
```

File: .cspell/cspell.json [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      "filename": "/core/modules/layout/tests/layout.test",
+      "ignoreWords": ["isnt"]
+    },
+    {
+      "filename": "/core/modules/simpletest/tests/session*",
+      "ignoreWords": ["dont"]
+    },
+    {
+      "filename": "/core/themes/bartik/color/color.inc",
+      "ignoreWords": ["Blackdrop"]
```
Comment (klonos) [inline .cspell/cspell.json]: Not sure if we should be ignoring entire files. cspell seems to support a `cspell:disable-next-line` comment that we can use to ignore specific lines. Should we do that instead?

File: core/modules/layout/tests/layout.test [hunk 3]
```diff
@@ -1981,9 +1981,7 @@ class LayoutBlockTest extends BackdropWebTestCase {
     $hero_title_1 = 'Wonderwoman';
     $hero_content_1 = 'Youth is your greatest weapon, your greatest tool.';
     $hero_title_2 = 'Black Widow';
-    // "isnt" is purposefully missing an apostrophe as it was causing tests to
-    // fail.
-    $hero_content_2 = "This isnt gonna end well...";
+    $hero_content_2 = "Whatever it takes.";
```
Comment (klonos) [inline core/modules/layout/tests/layout.test:1984]: I had a look in https://everydaypower.com/black-widow-quotes, and I'd like to suggest this instead:
```suggestion
    $hero_content_2 = "At some point, we all have to choose between what the world wants you to be and who you are.";
```

Comment (klonos) [reply core/modules/layout/tests/layout.test:1984]: ...or this shorter one:
```suggestion
    $hero_content_2 = "It was real to me too.";
```

Comment (ghost) [reply core/modules/layout/tests/layout.test:1984]: I don't think we really need to bikeshed on the wording for block content *in a test*. Request denied 😝

**Review by klonos — APPROVED** [review]
LGTM 👍🏼

**Comment by quicksketch** [comment]
Merged into 1.x and 1.24.x. Though PRs usually go against 1.x, I want to check that 1.24.x releases also exclude the `.cspell` directory as expected.

**Pull request merged**